tools/shared.py

```diff
@@ -1240,7 +1241,8 @@ def is_exe(fpath):
     if fpath:
       if is_exe(program): return program
     else:
-      for path in os.environ["PATH"].split(os.pathsep):
+      paths = os.environ["PATH"].split(os.pathsep) + (extra_paths or [])
```

Comment:
Avoiding putting the initializer in the default argument avoids the classic surprise about its semantics (e.g. http://stackoverflow.com/questions/1132941/least-astonishment-and-the-mutable-default-argument ). It's not a problem in this particular function but better to avoid it in general.